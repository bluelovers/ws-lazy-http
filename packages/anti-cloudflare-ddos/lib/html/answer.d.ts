/// <reference types="node" />
export declare function _getCloudflareKey(body: string | Buffer): string;
export declare function getCloudflareAnswer(body: string | Buffer, domain: string): string;
export default getCloudflareAnswer;
