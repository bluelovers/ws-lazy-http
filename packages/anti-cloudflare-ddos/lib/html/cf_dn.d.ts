/**
 * Created by user on 2020/5/1.
 */
/// <reference types="node" />
export declare function getCloudflareCFDNKey(body: string | Buffer): string;
export default getCloudflareCFDNKey;
