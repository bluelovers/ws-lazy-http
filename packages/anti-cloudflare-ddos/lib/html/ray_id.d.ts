/// <reference types="node" />
export declare function getCloudflareRayID(body: string | Buffer): string;
export default getCloudflareRayID;
