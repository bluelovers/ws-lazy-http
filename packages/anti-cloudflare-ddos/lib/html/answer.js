"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCloudflareAnswer = exports._getCloudflareKey = void 0;
function _getCloudflareKey(body) {
    var _a;
    const kRe = /\bk\s*=\s*'(cf-[\w-]+)'\s*;/g;
    return (_a = kRe.exec(String(body))) === null || _a === void 0 ? void 0 : _a[1];
}
exports._getCloudflareKey = _getCloudflareKey;
function getCloudflareAnswer(body, domain) {
    // @ts-ignore
    var s, t, o, p, b, r, e, a, k, i, n, g, f, hrewiEc = { "p": +((!+[] + !![] + !![] + !![] + !![] + !![] + !![] + !![] + !![] + []) + (!+[] + !![] + !![] + !![]) + (+!![]) + (+[]) + (!+[] + !![] + !![] + !![] + !![] + !![] + !![] + !![]) + (!+[] + !![] + !![]) + (!+[] + !![] + !![] + !![] + !![] + !![] + !![] + !![] + !![]) + (!+[] + !![] + !![] + !![] + !![] + !![] + !![] + !![]) + (+!![])) / +((!+[] + !![] + !![] + !![] + !![] + !![] + []) + (!+[] + !![] + !![] + !![] + !![] + !![] + !![] + !![]) + (!+[] + !![] + !![] + !![] + !![] + !![]) + (+!![]) + (!+[] + !![] + !![] + !![] + !![] + !![] + !![] + !![] + !![]) + (!+[] + !![] + !![] + !![]) + (!+[] + !![]) + (!+[] + !![]) + (!+[] + !![] + !![] + !![] + !![] + !![])) };
    g = String.fromCharCode;
    o = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=";
    e = function (s) {
        s += "==".slice(2 - (s.length & 3));
        var bm, r = "", r1, r2, i = 0, r3 = '<span class="cf-error-code">1020</span>';
        for (; i < s.length;) {
            bm = o.indexOf(s.charAt(i++)) << 18 | o.indexOf(s.charAt(i++)) << 12
                | (r1 = o.indexOf(s.charAt(i++))) << 6 | (r2 = o.indexOf(s.charAt(i++)));
            r += r1 === 64 ? g(bm >> 16 & 255)
                : r2 === 64 ? g(bm >> 16 & 255, bm >> 8 & 255)
                    : g(bm >> 16 & 255, bm >> 8 & 255, bm & 255);
        }
        return r;
    };
    t = domain;
    k = _getCloudflareKey(body);
    // @ts-ignore
    ;
    hrewiEc.p *= +((!+[] + !![] + !![] + !![] + !![] + !![] + !![] + !![] + !![] + []) + (!+[] + !![] + !![] + !![]) + (+!![]) + (+[]) + (!+[] + !![] + !![] + !![] + !![] + !![] + !![] + !![]) + (!+[] + !![] + !![]) + (!+[] + !![] + !![] + !![] + !![] + !![] + !![] + !![] + !![]) + (!+[] + !![] + !![] + !![] + !![] + !![] + !![] + !![]) + (+!![])) / +((!+[] + !![] + !![] + !![] + []) + (!+[] + !![] + !![]) + (!+[] + !![] + !![]) + (+[]) + (!+[] + !![] + !![] + !![] + !![]) + (!+[] + !![] + !![] + !![] + !![] + !![] + !![] + !![]) + (!+[] + !![] + !![]) + (!+[] + !![] + !![] + !![] + !![] + !![] + !![] + !![]) + (+[]));
    hrewiEc.p *= +((!+[] + !![] + !![] + !![] + !![] + !![] + !![] + !![] + []) + (+!![]) + (!+[] + !![] + !![] + !![] + !![] + !![] + !![]) + (!+[] + !![] + !![] + !![] + !![]) + (+[]) + (!+[] + !![] + !![] + !![]) + (!+[] + !![]) + (!+[] + !![] + !![] + !![] + !![]) + (!+[] + !![] + !![])) / (+(+((!+[] + !![] + !![] + []) + (!+[] + !![] + !![] + !![] + !![] + !![] + !![] + !![] + !![]) + (!+[] + !![]) + (!+[] + !![] + !![] + !![] + !![] + !![]) + (!+[] + !![] + !![] + !![] + !![] + !![] + !![] + !![] + !![]) + (+[]) + (!+[] + !![] + !![] + !![] + !![] + !![]) + (+!![]) + (!+[] + !![] + !![] + !![] + !![] + !![] + !![] + !![]))) + (function (p) { return eval((true + "")[0] + ".ch" + (false + "")[1] + (true + "")[1] + Function("return escape")()(("")["italics"]())[2] + "o" + (undefined + "")[2] + (true + "")[3] + "A" + (true + "")[0] + "(" + p + ")"); }(+((!+[] + !![] + !![] + !![] + !![] + !![] + [])))));
    hrewiEc.p -= +((!+[] + !![] + !![] + !![] + !![] + !![] + !![] + !![] + []) + (+!![]) + (!+[] + !![] + !![] + !![] + !![] + !![] + !![]) + (!+[] + !![] + !![] + !![] + !![]) + (+[]) + (!+[] + !![] + !![] + !![]) + (!+[] + !![]) + (!+[] + !![] + !![] + !![] + !![]) + (!+[] + !![] + !![])) / +((!+[] + !![] + !![] + !![] + []) + (!+[] + !![] + !![]) + (!+[] + !![] + !![] + !![] + !![] + !![] + !![] + !![] + !![]) + (!+[] + !![]) + (!+[] + !![] + !![] + !![] + !![]) + (!+[] + !![]) + (!+[] + !![] + !![] + !![] + !![] + !![]) + (!+[] + !![] + !![] + !![] + !![] + !![]) + (!+[] + !![] + !![] + !![] + !![] + !![] + !![]));
    let value = (+hrewiEc.p).toFixed(10);
    return value;
}
exports.getCloudflareAnswer = getCloudflareAnswer;
exports.default = getCloudflareAnswer;
//# sourceMappingURL=answer.js.map