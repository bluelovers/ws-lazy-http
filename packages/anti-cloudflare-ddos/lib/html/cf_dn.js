"use strict";
/**
 * Created by user on 2020/5/1.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCloudflareCFDNKey = void 0;
function getCloudflareCFDNKey(body) {
    var _a;
    const kRe = /\bk\s*=\s*'(cf-[\w-]+)'\s*;/g;
    return (_a = kRe.exec(String(body))) === null || _a === void 0 ? void 0 : _a[1];
}
exports.getCloudflareCFDNKey = getCloudflareCFDNKey;
exports.default = getCloudflareCFDNKey;
//# sourceMappingURL=cf_dn.js.map