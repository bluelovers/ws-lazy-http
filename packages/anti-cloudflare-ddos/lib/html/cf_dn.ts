/**
 * Created by user on 2020/5/1.
 */

export function getCloudflareCFDNKey(body: string | Buffer)
{
	const kRe = /\bk\s*=\s*'(cf-[\w-]+)'\s*;/g;
	return kRe.exec(String(body))?.[1];
}

export default getCloudflareCFDNKey
