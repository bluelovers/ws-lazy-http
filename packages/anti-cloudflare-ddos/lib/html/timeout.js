"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCloudflareTimeout = exports.timeoutReg = void 0;
exports.timeoutReg = /setTimeout\(\s*function\s*\(\){\s*(var\s+(?:[a-z]\s*,\s*)(?:[a-z]\s*,\s*)+(?:[a-z]\s*,\s*).+?\r?\n[\s\S]+?a\.value\s*=.+?)\r?\n[\s\S]+?,\s*(\d{4,5})\);\s*(?:\/\*eoc\*\/)?/;
function getCloudflareTimeout(body, defaultTimeout = 5000) {
    const timeout = exports.timeoutReg.exec(String(body));
    let ms = (timeout === null || timeout === void 0 ? void 0 : timeout.length) > 1 ? Number(timeout[2]) : defaultTimeout;
    if (ms <= 0) {
        ms = defaultTimeout;
    }
    return ms;
}
exports.getCloudflareTimeout = getCloudflareTimeout;
exports.default = getCloudflareTimeout;
//# sourceMappingURL=timeout.js.map