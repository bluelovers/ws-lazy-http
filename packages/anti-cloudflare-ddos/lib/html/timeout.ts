
export const timeoutReg = /setTimeout\(\s*function\s*\(\){\s*(var\s+(?:[a-z]\s*,\s*)(?:[a-z]\s*,\s*)+(?:[a-z]\s*,\s*).+?\r?\n[\s\S]+?a\.value\s*=.+?)\r?\n[\s\S]+?,\s*(\d{4,5})\);\s*(?:\/\*eoc\*\/)?/;

export function getCloudflareTimeout(body: string | Buffer, defaultTimeout = 5000)
{
	const timeout = timeoutReg.exec(String(body));

	let ms = timeout?.length > 1 ? Number(timeout[2]) : defaultTimeout;

	if (ms <= 0)
	{
		ms = defaultTimeout;
	}

	return ms
}

export default getCloudflareTimeout
