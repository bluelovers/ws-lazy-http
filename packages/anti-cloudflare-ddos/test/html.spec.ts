import { readFileSync } from "fs";
import { join, basename, parse } from 'path';
import { getCloudflareTimeout } from '../lib/html/timeout';
import getCloudflareAnswer from '../lib/html/answer';
import FastGlob from '@bluelovers/fast-glob/bluebird';
import getCloudflareRayID from '../lib/html/ray_id';

const __res = join(__dirname, 'res');

const htmlList = FastGlob.sync<string>([
		'*/*.html',
	], {
		cwd: __res,
	})
	.map(filename =>
	{

		let fp = parse(filename);

		return {
			name: filename,
			domain: fp.base.match(/^\d+_(.+)$/)[1],
			body: readFileSync(join(__res, filename)),
		}
	})
;

describe(`htmlList`, () =>
{
	htmlList
		.forEach((entry) =>
		{

			describe(entry.name, () =>
			{

				it(`getCloudflareTimeout`, () =>
				{

					let actual = getCloudflareTimeout(entry.body, 6000);

					expect(actual).toStrictEqual(4000);
					expect(actual).toMatchSnapshot();

				});

				it(`getCloudflareRayID`, () =>
				{

					let actual = getCloudflareRayID(entry.body);
					expect(actual).toMatchSnapshot();

				});

				it(`getCloudflareAnswer`, () =>
				{

					let actual = getCloudflareAnswer(entry.body, entry.domain);
					expect(actual).toBeDefined();

				});

			});

		})
	;
});

