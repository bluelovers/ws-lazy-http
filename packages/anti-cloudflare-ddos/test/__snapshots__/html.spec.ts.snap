// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`htmlList 2020/001_masiro.moe.html getCloudflareRayID 1`] = `"58b0c928fa0fefa6"`;

exports[`htmlList 2020/001_masiro.moe.html getCloudflareTimeout 1`] = `4000`;

exports[`htmlList 2020/002_masiro.moe.html getCloudflareRayID 1`] = `"58c7440a8bfa0e76"`;

exports[`htmlList 2020/002_masiro.moe.html getCloudflareTimeout 1`] = `4000`;

exports[`htmlList 2020/002_rlsbb.ru.html getCloudflareRayID 1`] = `"58c743c03a1e199f"`;

exports[`htmlList 2020/002_rlsbb.ru.html getCloudflareTimeout 1`] = `4000`;
